create_shards.c

```diff
@@ -73,6 +80,11 @@ master_create_distributed_table(PG_FUNCTION_ARGS)
 	AttrNumber partitionColumnId = InvalidAttrNumber;
 	char *partitionColumnName = text_to_cstring(partitionColumnText);
 	char *tableName = text_to_cstring(tableNameText);
+	Var* partitionColumnVar = NULL;
+	Oid partitionColumnTypeId = InvalidOid;
+	Oid partitionColumnOpClassId = InvalidOid;
+	Oid btreeSupportRoutine = InvalidOid;
```

Comment:
Move the declaration of `btreeSupportRoutine` and `hashSupportRoutine` into the respective `if`/`else if` blocks to reduce their scope… no reason to declare them up here. And when you move them, you can combine the declaration and assignment into a one-line definition for each.